prebuild/emscripten/glue.cpp

```diff
@@ -0,0 +1,538 @@
+  delete self;
+}
+
+// SecretKey
+
+blst::SecretKey* EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_SecretKey_0() {
+  return new blst::SecretKey();
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_keygen_3(blst::SecretKey* self, const blst::byte* ikm, long ikm_len, const char* info) {
+  self->keygen(ikm, ikm_len, info);
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_from_bendian_1(blst::SecretKey* self, const blst::byte* input) {
+  blst::byte _in[32];
+  for (int i = 0; i < 32; i++) {
+    _in[i] = input[i];
+  }
+  self->from_bendian(_in);
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_from_lendian_1(blst::SecretKey* self, blst::byte* input) {
+  blst::byte _in[32];
+  for (int i = 0; i < 32; i++) {
+    _in[i] = input[i];
+  }
+  self->from_lendian(_in);
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_to_bendian_1(blst::SecretKey* self, blst::byte* out) {
+  blst::byte _out[32];
+  self->to_bendian(_out);
+  for (int i = 0; i < 32; i++) {
+    out[i] = _out[i];
+  }
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey_to_lendian_1(blst::SecretKey* self, blst::byte* out) {
+  blst::byte _out[32];
+  self->to_lendian(_out);
+  for (int i = 0; i < 32; i++) {
+    out[i] = _out[i];
+  }
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_SecretKey___destroy___0(blst::SecretKey* self) {
+  delete self;
+}
+
+// PT
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_PT__P1_Affine_1(const blst::P1_Affine* p) {
+  return new blst::PT(*p);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_PT__P2_Affine_1(const blst::P2_Affine* p) {
+  return new blst::PT(*p);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_PT_2(const blst::P1_Affine* p, const blst::P2_Affine* q) {
+  return new blst::PT(*p, *q);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_dup_0(blst::PT* self) {
+  blst::PT *dup = (blst::PT *) malloc(sizeof(blst::PT));
+  new (dup) blst::PT(self->dup());
+  return dup;
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_is_one_0(blst::PT* self) {
+  return self->is_one();
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_is_equal_1(blst::PT* self, const blst::PT* p) {
+  return self->is_equal(*p);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_sqr_0(blst::PT* self) {
+  return self->sqr();
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_mul_1(blst::PT* self, const blst::PT* p) {
+  return self->mul(*p);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_final_exp_0(blst::PT* self) {
+  return self->final_exp();
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_in_group_0(blst::PT* self) {
+  return self->in_group();
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_finalverify_2(blst::PT* self, const blst::PT* gt1, const blst::PT* gt2) {
+  return self->finalverify(*gt1, *gt2);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_one_0(blst::PT* self) {
+  blst::PT *one = (blst::PT *) malloc(sizeof(blst::PT));
+  *one = self-> one();
```

Comment:
This is potentially a **big explosion** here!

The whole thing should be able to be done with:

```c++
blst::PT *one = new blst::PT(self->one());
```
Assuming `blst::PT` has a copy constructor